tests/test_server.py

```diff
@@ -1893,3 +1893,140 @@ async def put_version(version: int) -> int:
     # give it time to attempt to be release
     await asyncio.sleep(0.1)
     assert not await clienthelper.is_released(v1)
+
+
+async def test_set_fact(
+    server,
+    client,
+    clienthelper,
+    environment,
+    no_agent_backoff,
+    agent_factory,
+):
+    """
+    Test the set_fact endpoint. First create a fact with expires set to true.
+    Then set expires to false for the same fact.
+    """
+    await agent_factory(
+        environment=environment,
+        agent_map={"discovery_agent": "localhost"},
+        hostname="discovery_agent",
+    )
```

Comment:
Is there a reason why this agent needs to be started. If not, I wouldn't do it, because it only makes the test case slower.